packages/flutter/lib/src/material/material_state.dart

```diff
@@ -104,6 +104,11 @@ typedef MaterialPropertyResolver<T> = T Function(Set<MaterialState> states);
 /// to provide a `defaultValue` to the super constructor, so that we can know
 /// at compile-time what its default color is.
 ///
+/// This class enables existing widget implementations with [Color]
+/// properties to be extended to also effectively support `MaterialStateProperty<Color>`
+/// property values. [MaterialStateColor] should only be used with widgets that document
+/// their support, like [TimePickerTheme].
```

Comment:
Best to link to `[TimePickerThemeData.dayPeriodColor]` to the reader ends up in an example that uses MaterialStateColor.